lib/plugins/threaded-signing/plugin.c

```diff
@@ -111,18 +115,28 @@ local_signature_info_create(const signature_info_t *signature_info)
 }
 
 /* Frees all allocated memory and resets members. Excluded are the worker thread members |thread|,
- * |mutex|, |cond| and |is_running|, but also the counter |nbr_of_unsigned_hashes|. */
+ * |mutex|, |cond| and |is_running|. */
 static void
 sv_threaded_plugin_reset(sv_threaded_plugin_t *self)
 {
   local_signature_info_free(self->signature_info);
   self->signature_info = NULL;
-  free(self->hash_to_sign);
-  self->hash_to_sign = NULL;
+
+  for (int i = 0; i < MAX_HASHES; i++) {
+    free(self->output_buffer[i].signature);
+    self->output_buffer[i].signature = NULL;
+    self->output_buffer[i].size = 0;
```

Comment:
To be fully correct, and minimize potential bugs, you should probably also set `threaded_signing_error = false;`